### PR TITLE
Check rights retention before oa works

### PR DESCRIPTION
**Pull request by anusharanganathan**
PR for https://github.com/antleaf/jct-project/issues/503

### CODE REVIEW

File: server/service/jct/api.coffee [hunk 1]
```diff
@@ -787,51 +783,54 @@ API.service.jct.permission = (issn, institution) ->
...
-      rs.qualifications.push(q)
-  rs.log ?= []
-  if res_sa.log? and res_sa.log.length
-    for l in (if _.isArray(res_sa.log) then res_sa.log else [res_sa.log])
-      rs.log.push(l)
+  # merge the qualifications and logs from SA prohibition and SA rights retention into OA.Works permission
+  rs = _merge_logs_and_qa(res_r, rs)
 
   # check for retention
   if rs
```
Comment (richard-jones) [inline server/service/jct/api.coffee:819]: It looks to me like rs is always truthy

File: server/service/jct/api.coffee [hunk 2]
```diff
@@ -787,51 +783,54 @@ API.service.jct.permission = (issn, institution) ->
...
     _rtn = {}
```
Comment (richard-jones) [inline server/service/jct/api.coffee:820]: This variable never gets used

File: server/service/jct/api.coffee [hunk 3]
```diff
@@ -787,51 +783,54 @@ API.service.jct.permission = (issn, institution) ->
...
     for r in (if _.isArray(rs) then rs else [rs])
```
Comment (richard-jones) [inline server/service/jct/api.coffee:821]: I'm struggling to follow the logic in this section, but looking at the workflow diagram it seems like we can make a shortcut here in that if the OAB permissions check comes back as compliant, then we can return the result straight away, without going through this retention section first

File: server/service/jct/api.coffee [hunk 4]
```diff
@@ -787,51 +783,54 @@ API.service.jct.permission = (issn, institution) ->
...
-            r.log.push(lg) for lg in _rtn[journal].log
-            if _rtn[journal].qualifications? and _rtn[journal].qualifications.length
-              r.qualifications ?= []
-              r.qualifications.push(ql) for ql in _rtn[journal].qualifications
+            # 22022022, as per https://github.com/antleaf/jct-project/issues/503
+            # Add rights retention author qualification
+            r.qualifications ?= []
+            r.qualifications.push({'rights_retention_author_advice': ''})
           else
             r.log.push code: 'SA.FunderRRNotActive'
```
Comment (richard-jones) [inline server/service/jct/api.coffee:835]: After SA.FunderRRNotActive is triggered, the next stage in the SA workflow says to add SA.Unknown or SA.NotCompliant based on what the result of the OAB check was.  Perhaps this is being done elsewhere using that merge code above, but I can't see it easily.

**Review by richard-jones — CHANGES_REQUESTED** [review]

File: server/service/jct/api.coffee (hunk 3 above)
Comment (anusharanganathan) [reply server/service/jct/api.coffee]: Discussed and understood current code

File: server/service/jct/api.coffee (hunk 4 above)
Comment (anusharanganathan) [reply server/service/jct/api.coffee]: Implemented this

**Review by anusharanganathan — COMMENTED** [review]
Went through all of the comments and made relevant changes

**Pull request merged**